src/main/java/com/example/producerconsumer/QueueConsumerImpl.java

```diff
@@ -42,10 +47,13 @@ public void consume(String operateOn) {
         // and consumer.consume would still work correctly but producers could start little earlier and the prints to console could be inconsistent
         // ...and if there were no 'half of the queue is empty' condition at all then using Spliterator and Stream's would be much cleaner approach.
         synchronized (pcJobContext.getWaitForEmptyingHalfOfTheQueue()) {
+            //dlaczego czekamy aż połoa kolejki będzie pusta? Jaki to będzie miało wpływ gdy kolejka będzie miała np 10_000 elementów - czekamy aż 5_000 będzie skonsumowane?
 
```

Comment:
Wydawało mi się że takie jest wymaganie w tym zadaniu a dokłądnie w zdaniu: "...Once the queue is full TaskProducer stops producing tasks and waits (periodically polling the queue) until its size drops to half of the maximum size and starts producing tasks again."

Reply:
OK, przeoczyłem - nie ta wersja zadania :)